elf_binary.cc

```diff
@@ -572,6 +572,11 @@ Elf_Addr ELFBinary::OffsetFromAddr(Elf_Addr addr) const {
             return addr - phdr->p_vaddr + phdr->p_offset;
         }
     }
+    if (tls() != nullptr) {
+        if (tls()->p_vaddr <= addr && addr < tls()->p_vaddr + tls()->p_memsz) {
```

Comment:
Isn't `if (tls()->p_vaddr == addr) return tls()->p_offset;` enough? Offsets would not make sense if `tls()->p_vaddr != addr`?

Reply:
Yes, it's enough. https://github.com/shinh/sold/pull/76/commits/6aa5625d0b949758710ed58d2e61d0ad82634d07